Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.h

```diff
@@ -174,25 +174,26 @@ class  VTK_SLICER_MARKUPS_MODULE_MRML_EXPORT vtkMRMLMarkupsCurveNode : public vt
   void SetCurveTypeToCardinalSpline();
   void SetCurveTypeToKochanekSpline();
   void SetCurveTypeToPolynomial();
-  void SetCurveTypeToShortestSurfaceDistance(vtkMRMLModelNode* modelNode=nullptr);
+  void SetCurveTypeToShortestDistanceOnSurface(vtkMRMLModelNode* modelNode=nullptr);
 
   /// Node reference role for the surface that is used in the shortest surface distance curve type
-  const char* GetSurfaceMeshNodeReferenceRole() { return "surfaceMesh"; };
-  const char* GetSurfaceMeshNodeReferenceMRMLAttributeName() { return "surfaceMeshRef"; };
+  const char* GetShortestDistanceSurfaceNodeReferenceRole() { return "shortestDistanceSurface"; };
+  const char* GetShortestDistanceSurfaceNodeReferenceMRMLAttributeName() { return "shortestDistanceSurfaceRef"; };
 
-  /// The model node that is used for calculating the shortest surface distance curve type
-  void SetAndObserveModelNode(vtkMRMLModelNode* modelNode);
-  vtkMRMLModelNode* GetModelNode();
+  /// The model node that is used as the surface mesh for finding the shortest distance path on the surface mesh.
+  /// Used by the ShortestDistanceOnSurface curve type.
+  void SetAndObserveShortestDistanceSurfaceNode(vtkMRMLModelNode* modelNode);
+  vtkMRMLModelNode* GetShortestDistanceSurfaceNode();
 
-  /// Whether to scale the distance of the points by some scalar weight value.
-  /// Only applies to shortest surface distance curve type.
-  bool GetUseSurfaceScalarWeights();
-  void SetUseSurfaceScalarWeights(bool useSurfaceScalarWeights);
+  /// The method that should be used to combine the distance with the scalar value.
+  /// Uses the COST_FUNCTION_X enums from vtkSlicerDijkstraGraphGeodesicPath.
+  int GetSurfaceCostFunction();
+  void SetSurfaceCostFunction(int surfaceCostFunction);
```

Comment:
The current name suggests that it is a function (similar to SurfaceScalarWeightFunction, which is a string). Adding "Type" in the name (SurfaceCostFunctionType) would be better.